ext/compressors/iaa/iaa_compress.c

```diff
@@ -0,0 +1,208 @@
+    if(result_len > 0) {
+        *compression_failed = 0;
+        *result_lenp = (size_t)result_len;
+        return (0);
+    }
+    *compression_failed = 1;
+    return (iaa_error(compressor, session, "iaa_compress", -1));
+}
+
+/*
+ * iaa_decompress --
+ *     WiredTiger iaa decompression.
+ */
+static int
+iaa_decompress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size_t src_len,
+  uint8_t *dst, size_t dst_len, size_t *result_lenp)
+{
+    uint32_t result_len = 0;
+    int ret;
+    doDecompressData(compressor, session, src, (uint32_t)src_len, dst, (uint32_t)dst_len, &result_len);
+    if(result_len > 0) {
+        *result_lenp = (size_t)result_len;
+        ret = 0;
+    } else {
+        ret = -1;
+    }
+    return (ret == 0 ? 0 : iaa_error(compressor, session, "iaa_decompress", ret));
+}
+
+/*
+ * iaa_terminate --
+ *     WiredTiger iaa compression termination.
+ */
+static int
+iaa_terminate(WT_COMPRESSOR *compressor, WT_SESSION *session)
+{
+    (void)session; /* Unused parameters */
+
+    free(compressor);
+    return (0);
+}
+
+/*
+ * iaa_pre_size --
+ *     WiredTiger iaa destination buffer sizing.
+ */
+static int
+iaa_pre_size(
+  WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, size_t src_len, size_t *result_lenp)
+{
+    (void)compressor; /* Unused parameters */
+    (void)session;
+    (void)src;
+
+    /*
+     * iaa requires the dest buffer be somewhat larger than the source. Fortunately, this is fast
+     * to compute, and will give us a dest buffer in iaa_compress that we can compress to
+     * directly.
+     */
+    *result_lenp = getMaxCompressedDataSize((uint32_t)src_len);;
+    return (0);
+}
+
+/*
+ * iaa_add_compressor --
+ *     Add a iaa compressor.
+ */
+static int
+iaa_add_compressor(WT_CONNECTION *connection, const char *name)
+{
+    iaa_COMPRESSOR *iaa_compressor;
+    int ret;
+    if ((iaa_compressor = calloc(1, sizeof(iaa_COMPRESSOR))) == NULL)
+        return (errno);
+
+    iaa_compressor->compressor.compress = iaa_compress;
+    iaa_compressor->compressor.decompress = iaa_decompress;
+    iaa_compressor->compressor.pre_size = iaa_pre_size;
+    iaa_compressor->compressor.terminate = iaa_terminate;
+
+    iaa_compressor->wt_api = connection->get_extension_api(connection);
+
+    /* Load the compressor. */
+    if ((ret = connection->add_compressor(
+           connection, name, (WT_COMPRESSOR *)iaa_compressor, NULL)) == 0) {
+	return (0);
+    }
+
+    free(iaa_compressor);
+    return (ret);
+}
+
+
+int iaa_extension_init(WT_CONNECTION *, WT_CONFIG_ARG *);
+
+/*
+ * iaa_extension_init --
+ *     WiredTiger iaa compression extension - called directly when iaa support is built in, or via
+ *     wiredtiger_extension_init when iaa support is included via extension loading.
+ */
+int
+iaa_extension_init(WT_CONNECTION *connection, WT_CONFIG_ARG *config)
+{
+    int ret;
+    (void)config;
+
+    if ((ret = iaa_add_compressor(connection, "iaa")) != 0)
```

Comment:
We can simplify here:
```suggestion
    ret = iaa_add_compressor(connection, "iaa"));
    return (ret)
```

Reply:
Good!
We will modify it. Thanks!